File: sqfile/src/main/java/com/effective/android/sqfile/anno/ConflictStrategy.java
package com.effective.android.sqfile.anno;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.SOURCE;

@Retention(SOURCE)
@IntDef({ConflictStrategy.REPLACE, ConflictStrategy.ABORT, ConflictStrategy.ERROR})
public @interface ConflictStrategy {
    int REPLACE = 1;        //替换
    int ABORT = 2;          //中止
    int ERROR = 3;          //异常
}

File: sqfile/src/main/java/com/effective/android/sqfile/param/anno/ConflictStrategy.kt
package com.effective.android.sqfile.param.anno

import androidx.annotation.IntDef

@Retention(AnnotationRetention.SOURCE)
@IntDef(ConflictStrategy.REPLACE, ConflictStrategy.ABORT, ConflictStrategy.ERROR)
annotation class ConflictStrategy {
    companion object {
        const val REPLACE = 1   //覆盖
        const val ABORT = 2     //中止
        const val ERROR = 3     //异常
    }
}